wp-content/plugins/goonj-blocks/build/functions.php

```diff
@@ -15,4 +15,208 @@ function gb_format_time_range($start_date, $end_date, $format = 'h:i A') {
+    } catch (\Exception $e) {
+        \Civi::log()->error('Error generating induction slots', [
+            'contactId' => $contactId,
+            'error' => $e->getMessage()
+        ]);
+        return [];
+    }
+}
+
+/**
+ * Generate specific induction slots based on induction type and scheduled activities.
+ *
+ * @param int $assignedOfficeId The office ID where slots are generated.
+ * @param int $maxSlots The maximum number of slots to generate.
+ * @param array $scheduledActivities List of already scheduled induction activities.
+ * @param string $inductionType The type of induction ('Processing_Unit' or 'Online_only_selected_by_Urban_P').
+ * @param DateTime $startDate The start date for generating slots.
+ * @return array Array of generated slots with day, date, time, count, and type.
+ */
+function generate_slots($assignedOfficeId, $maxSlots, $inductionType, $startDate) {
+    $slots = [];
+    $slotCount = 0;
+    $highActivityCountDays = 0;
+
+    try {
+        // Fetch office induction details for induction scheduling
+        $officeDetails = \Civi\Api4\Contact::get(FALSE)
+            ->addSelect('display_name', 'Goonj_Office_Details.Physical_Induction_Slot_Days:name', 'Goonj_Office_Details.Physical_Induction_Slot_Time', 'Goonj_Office_Details.Online_Induction_Slot_Days:name', 'Goonj_Office_Details.Online_Induction_Slot_Time')
+            ->addWhere('contact_sub_type', 'CONTAINS', 'Goonj_Office')
+            ->addWhere('id', '=', $assignedOfficeId)
+            ->execute()->first();
+        
+        if (empty($officeDetails) || empty($assignedOfficeId)) {
+            \Civi::log()->info('Office Details not found', ['assignedOfficeId' => $assignedOfficeId]);
+            return FALSE;
+        }
+
+        // Determine valid induction days and time based on induction type
+        $validInductionDays = ($inductionType === 'Processing_Unit')
+            ? $officeDetails['Goonj_Office_Details.Physical_Induction_Slot_Days:name']
+            : $officeDetails['Goonj_Office_Details.Online_Induction_Slot_Days:name'];
+
+        $inductionTime = ($inductionType === 'Processing_Unit')
+            ? $officeDetails['Goonj_Office_Details.Physical_Induction_Slot_Time']
+            : $officeDetails['Goonj_Office_Details.Online_Induction_Slot_Time'];
+
+        list($hour, $minute) = explode(':', $inductionTime);
+
+        // Fetch already scheduled induction activities at the assigned Goonj office
+        $scheduledActivities = \Civi\Api4\Activity::get(FALSE)
+            ->addSelect('activity_date_time')
+            ->addWhere('activity_type_id:name', '=', 'Induction')
+            ->addWhere('status_id:name', '=', 'Scheduled')
+            ->addWhere('Induction_Fields.Goonj_Office', '=', $assignedOfficeId)
+            ->addWhere('activity_date_time', '>', (new DateTime('today midnight'))->format('Y-m-d H:i:s'))
+            ->setLimit(60)
```

Comment:
Why we needed this, are we fetching only 60 scheduled induction activities at the assigned Goonj office?

Reply:
removed this